### DISCUSSION TITLE
MorganGenerator: bitInfo (and other fingerprints)

### DISCUSSION THREAD

**Discussion by RuudFirsa** [rdkit/rdkit#8041] [Q&A]
While it's appreciated that code improvements are continuously made, some changes may go to far. 

Let's consider the old syntax of getting the bitinfo in morgan fingerprints, one had to conveniently write three lines of code:
from rdkit.Chem.rdMolDescriptors import GetMorganFingerprint as MFP
bitInfo = {}
res = MFP(Chem.MolFromSmiles(<smiles>), radius, bitInfo=bitInfo)

The old method was a very effective method compared to the new method with all the freaky AdditionalOutput-instances.
Can the new method produce the bitInfo that was available in the old method, and if so, what syntax to use (as it's not obvious at all from the documentation)? 

Thanks, Ruud

**Comment by bp-kelley** [comment]
There is a good blog post about the changes here:

https://greglandrum.github.io/rdkit-blog/posts/2023-01-18-fingerprint-generator-tutorial.html

The main difference is that an output object now contains the various data a fingerprint can generate as opposed to the passed in variables.

Here is an example usage:

```
ao = rdFingerprintGenerator.AdditionalOutput()
ao.AllocateBitInfoMap()

fp1 = rdFingerprintGenerator.GetMorganGenerator(radius=3).GetFingerprint(m, additionalOutput=ao)
print(ao.GetBitInfoMap())
```

Hope this helps.

**Comment by RuudFirsa** [comment]
confidential

Thanks.

The tricky part is the ao.AllocateBitInfoMap()
But yeah, it with GetSparseFingerprint it gives what I’m expecting.
***@***.***




From: Brian Kelley ***@***.***>
Date: Monday, 25 November 2024 at 16:38
To: rdkit/rdkit ***@***.***>
Cc: Ruud van Deursen ***@***.***>, Author ***@***.***>
Subject: Re: [rdkit/rdkit] MorganGenerator: bitInfo (and other fingerprints) (Discussion #8041)

CAUTION: This email originated from outside dsm-firmenich. DO NOT CLICK on links and DO NOT OPEN attachments unless you recognize the sender and know the content is safe. If you think it is suspicious, report this email using Report Suspicious Email button available (once the email is opened) in the Microsoft Outlook ribbon above.


There is a good blog post about the changes here:

https://greglandrum.github.io/rdkit-blog/posts/2023-01-18-fingerprint-generator-tutorial.html<https://urldefense.com/v3/__https:/greglandrum.github.io/rdkit-blog/posts/2023-01-18-fingerprint-generator-tutorial.html__;!!A4qNoLW32FtaT8k!QI8D0Zi2oNaOsuaBW-xjpyIj1bN2XTwM7mIlQi1AwNyXGN9L_iyoHjbrbR21cV7GZnHq6BDoWFX04JbXWkgPAWDRxD5I3hM23B3q$>

The main difference is that an output object now contains the various data a fingerprint can generate as opposed to the passed in variables.

Here is an example usage:

ao = rdFingerprintGenerator.AdditionalOutput()

ao.AllocateBitInfoMap()



fp1 = rdFingerprintGenerator.GetMorganGenerator(radius=3).GetFingerprint(m, additionalOutput=ao)

print(ao.GetBitInfoMap())

Hope this helps.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/rdkit/rdkit/discussions/8041*discussioncomment-11374145__;Iw!!A4qNoLW32FtaT8k!QI8D0Zi2oNaOsuaBW-xjpyIj1bN2XTwM7mIlQi1AwNyXGN9L_iyoHjbrbR21cV7GZnHq6BDoWFX04JbXWkgPAWDRxD5I3shNFWYm$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/ALI2MZ6VLPTTMFNEFSYKEXD2CM75HAVCNFSM6AAAAABSNVRBZCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZXGQYTINI__;!!A4qNoLW32FtaT8k!QI8D0Zi2oNaOsuaBW-xjpyIj1bN2XTwM7mIlQi1AwNyXGN9L_iyoHjbrbR21cV7GZnHq6BDoWFX04JbXWkgPAWDRxD5I3slGugyU$>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

______________________________________________________________________


DISCLAIMER:
This email and any files transmitted with it, including replies and forwarded copies subsequently transmitted from dsm-firmenich, are confidential and for the intended recipient only.
If you have received this email by mistake then please notify the sender immediately and then delete it from your system; access, disclosure, copying, distribution or reliance on any of it by anyone else is prohibited.
